jx/ui/JXMLComponent.js

```diff
@@ -127,13 +127,17 @@ JXMLComponent.prototype.setAttr = function(delta) {
 JXMLComponent.prototype.applyAttr = function(delta) {
 	if (this.root) { // component has internal structure
 		// Allow component to mangle attributes
-		if (this.resolved.onAttr)
-			this.resolved.onAttr(delta, this.attr);
+		if (this.resolved.onDelta)
+			this.resolved.onDelta(delta, this.attr);
 
-		JXML.deepMerge(this.attr, delta);
+		delta = JXML.mergeDiff(this.attr, delta);
+
+		if (this.resolved.onAttr && delta)
+			this.resolved.onAttr(this.attr, delta);
```

Comment:
No, the beauty of this is it only affects your root/children. This allows you to supply missing attributes but not overwrite the fact that originally, that attribute was not specified.

So for example, if a layout component automatically calculates a child's width, on the next pass, it will not think that the width was already specified and so fail to calculate the width.